.github/workflows/package.yml

```diff
@@ -0,0 +1,373 @@
+name: Package
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  merge_group:
+
+env:
+  TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
+  TURBO_TEAM: hashintel
+  TURBO_REMOTE_ONLY: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  setup:
+    runs-on: ubuntu-22.04
+    outputs:
+      units: ${{ steps.packages.outputs.units }}
+      integrations: ${{ steps.packages.outputs.integrations }}
+      dockers: ${{ steps.packages.outputs.dockers }}
+      lints: ${{ steps.packages.outputs.lints }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
+        with:
+          fetch-depth: 2
+
+      - name: Install turbo
+        run: yarn global add turbo
+
+      - name: Determine changed packages
+        id: packages
+        run: |
```

Comment:
Would it make sense for these large `run` statements to put them into separate `.sh` script files? I feel like that would increase maintainability quite drastically as we can debug/run/edit them locally before pushing and avoiding the painful push -> check -> change workflow.

Reply:
I was planning on having a separate script file for this but will do that in a follow-up.  I'd prefer to have the script either in another language or adding shellckeck to the CI.